### DISCUSSION TITLE
Is multi device need to run multi time of tag parser?

### DISCUSSION THREAD

**Discussion by asd055189** [microsoft/vscode-cpptools#9873] [Q&A]
Hello, I am currently tracing a large project, The tag parser takes me a long time to resolve the include path.
I have 3 devices, which are Macbook, and two windows desktops.
My question is, do I need to run the `tag parser` every time when I start a new session on my device?
Is there sth like a result file that I can clone into every device?
I already Google it, but still can't find the answer.
Can anyone help me?

**Comment by sean-mcmanus** [comment]
In general, there is no way to transfer the database file to another device -- all the paths stored in the database would have to exist, which definitely won't work between Mac and Windows. You could try copying over the database file from one Windows machine to another, but I'm not sure if it'll work or not. You may also be able to speed up the tag parsing by adding certain folders to the C_Cpp.files.exclude setting.

The database created from tag parsing is supposed to be preserved across sessions, but there are cases where this may not happen (https://github.com/microsoft/vscode-cpptools/issues/9837).

**Reply by asd055189** [reply]
I see, another question is that actually I use these devices to connect to a remote server, after reading some documents I create a `c_cpp_propertios.json` as follows under `${workspaceFolder}/.vscode`

{
    "configurations": [
        {
            "name": "Linux",
            "includePath": [
                "${workspaceFolder}/**"
            ],
            "defines": [],
            "compilerPath": "/opt/rh/devtoolset-9/root/usr/bin/gcc",
            "cStandard": "gnu17",
            "cppStandard": "gnu++14",
            "intelliSenseMode": "linux-gcc-x64",
            "browse": {
                "limitSymbolsToIncludedHeaders": true,
                "databaseFilename": "${workspaceFolder}/.vscode/browse.vc.db",
                "path": [
                    "${workspaceFolder}/*"
                ]
            }
        }
    ],
    "version": 4
}

I am wondering that is it possible that all of my devices can read this DB file using the remote ssh?
Because it seems like I am going to re-parser the file again when I start another session on another device

**Reply by sean-mcmanus** [reply]
You're using the VS Code remote extensions to connect to a remote machine? If that's the case, then our extension should be running on the remote machine and the database would be on the remote machine and shared by all the host machines. The source files are all on the remote machine, right?

**Reply by asd055189** [reply]
Yes, the source file and database are on the remote server.

This problem might be solved, it seems the database is not prepared until the parsing is done. If I restart the session before the parser finishes its work, the parser will recreate the database.

Thanks for the help. Maybe I should study `c_cpp_propertios.json` to boost the parser.

**Reply by sean-mcmanus** [reply]
The database will get reset if our cpptools process is not shutdown cleanly and forceably shutdown before 5 seconds have passed since the last database write. We've been getting multiple reports of the database resetting when it shouldn't...we're looking into ways to reduce this: https://github.com/microsoft/vscode-cpptools/issues/9837 .
